docs/classes.rst

```diff
@@ -210,6 +211,97 @@ There are already some utility classes implemented in ``pytest-mh``. See
     Each change that is made through the utility object (such as writing to a
     file) is automatically reverted (the original file is restored).
 
+TopologyController
+==================
+
+Topology controller can be assigned to a topology via `@pytest.mark.topology`
+or through known topology class. This controller provides various methods to
+control the topology behavior:
+
+* per-topology setup and teardown, called once before the first test/after the
+  last test for given topology is executed
+* per-test topology setup and teardown, called before and after every test case
+  for given topology
+* check topology requirements and skip the test if these are not satisfied
+
+In order to use the controller, you need to inherit from
+:class:`~pytest_mh.TopologyController` and override desired methods. Each method
+can take any parameter as defined by the topology fixtures. The parameter value
+is an instance of a :class:`~pytest_mh.MultihostHost` object.
+
+See :class:`~pytest_mh.TopologyController` for API documentation
+
+.. code-block:: python
+    :caption: Example topology controller
+
+    class ExampleController(TopologyController):
+        def skip(self, client: ClientHost) -> str | None:
+            result = client.ssh.run(
+                '''
+                # Implement your requirement check here
+                exit 1
+                ''', raise_on_error=False)
+            if result.rc != 0:
+                return "Topology requirements were not met"
+
+            return None
+
+        def topology_setup(self, client: ClientHost):
+            # One-time setup, prepare the host for this topology
+            # Changes done here are shared for all tests
+            pass
+
+        def topology_teardown(self, client: ClientHost):
+            # One-time teardown, this should undo changes from
+            # topology_setup
+            pass
+
+        def setup(self, client: ClientHost):
+            # Perform per-topology test setup
+            # This is called before execution of every test
+            pass
+
+        def teardown(self, client: ClientHost):
+            # Perform per-topology test teardown, this should undo changes
+            # from setup
+            pass
+
+.. code-block:: python
+    :caption: Example with low-level topology mark
+
+    class ExampleController(TopologyController):
+        # Implement methods you are interested in here
+        pass
+
+    @pytest.mark.topology(
+        "example", Topology(TopologyDomain("example", client=1)),
+        controller=ExampleController(),
+        fixtures=dict(client="example.client[0]")
+    )
+    def test_example(client: Client):
+        pass
+
+.. code-block:: python
+    :caption: Example with KnownTopology (recommended)
+
+    class ExampleController(TopologyController):
+        # Implement methods you are interested in here
+        pass
+
+    @final
+    @unique
+    class KnownTopology(KnownTopologyBase):
```

Comment:
Comment here that `example` should be changed to the project appropriate domain id `sssd` ?

Reply:
No. This request comes from lack of knowledge, that is well explain in other parts of the documentation. It can not be mentioned everywhere. Doing this, I could also mention "use your or topology and topology domain, host names and classes, fixture names, your own code and test"... this is an example, it should be treated as such.

Reply:
Fair point, what helped me is re-reading the https://tests.sssd.io/en/latest/writing-tests.html#deep-dive-into-multihost-fixtures documentation